sympy/sets/sets.py

```diff
@@ -2027,6 +2027,61 @@ def __iter__(self):
                 yield item
 
 
+
+class DisjointUnion(Set):
+    """ Represents the disjoint union (also known as the external disjoint union)
+    of a finite number of sets.
+
+    Examples
+    ========
+
+    >>> from sympy import DisjointUnion, FiniteSet, Interval
+    >>> A = FiniteSet(1, 2, 3)
+    >>> B = Interval(0, oo)
+    >>> DisjointUnion(A, B)
+    DisjointUnion(FiniteSet(1, 2, 3), Interval(0, oo))
+    >>> DisjointUnion(A, B).rewrite(Union)
+    ({1, 2, 3} × {0}) ∪ ([0, ∞) × {1})
+    >>> C = FiniteSet(Symbol('x'), Symbol('y'), Symbol('z'))
+    >>> DisjointUnion(C, C)
+    DisjointUnion(FiniteSet(x, y, z), FiniteSet(x, y, z))
+    >>> DisjointUnion(C, C).rewrite(Union)
+    {x, y, z} × {0, 1}
+
+    References
+    ==========
+
+    https://en.wikipedia.org/wiki/Disjoint_union
+    """
+
+    def __new__(cls, *sets):
+        dj_collection = []
+        for set_i in sets:
+            if isinstance(set_i, Set):
+                dj_collection.append(set_i)
+            else:
+                raise ValueError("Invalid input: '%s', input args \
+                    to DisjointUnion must be Sets" % set_i)
+        obj = Basic.__new__(cls, *dj_collection)
+        return obj
+
+    def rewrite(self, obj):
+        if obj == Union:
```

Comment:
You don't need a new function `rewrite` in this because sympy objects already have.
You can move this under `_eval_rewrite_as_Union` and sympy automatically dispatches `rewrite(Union)` to there

Reply:
Okay, I'll do that.

Reply:
[https://github.com/sympy/sympy/blob/09b4cfae284e020e7f94f03e037fbf3b2c489c5d/sympy/sets/sets.py#L2068-L2078](url)
Done